lib/streamlit/runtime/fragment.py

```diff
@@ -194,6 +194,7 @@ def wrapped_fragment():
             # Set ctx.current_fragment_id so that elements corresponding to this
             # fragment get tagged with the appropriate ID. ctx.current_fragment_id gets
             # reset after the fragment function finishes running.
+            prev_fragment_id = ctx.current_fragment_id
```

Comment:
nit: we should update the comment above a little bit to capture how we landed here. Maybe sth. like: `# we reset the fragment id to the previously running fragment so that nesting works correctly`

Reply:
done!